src/app/shares/viewshareaccount/viewshareaccount.component.ts

```diff
@@ -0,0 +1,354 @@
+            {
+              name: 'Modify Application',
+              icon: 'fa fa-pencil '
+            },
+            {
+              name: 'Approve',
+              icon: 'fa fa-check'
+            }
+          ],
+          options: [
+            {
+              name: 'Reject'
+            },
+            {
+              name: 'Delete'
+            }
+          ]
+        };
+      }
+
+      if (data.status.value === 'Approved') {
+        this.buttons = {
+          singlebuttons: [
+            {
+              name: 'Undo Approval',
+              icon: 'fa fa-undo'
+            },
+            {
+              name: 'Activate',
+              icon: 'fa fa-check'
+            }
+          ]
+        };
+      }
+
+      if (data.status.value === 'Active') {
+        if (this.sharesPendingForApproval) {
+          this.buttons = {
+            singlebuttons: [
+              {
+                name: 'Apply Additional Shares',
+                icon: 'fa fa-arrow-right'
+              },
+              {
+                name: 'Approve Additional Shares',
+                icon: 'fa fa-arrow-right'
+              },
+              {
+                name: 'Reject Additional Shares',
+                icon: 'fa fa-arrow-right'
+              },
+              {
+                name: 'Redeem Shares',
+                icon: 'fa fa-arrow-left'
+              }
+            ],
+            options: [
+              {
+                name: 'Close'
+              }
+            ]
+          };
+        } else {
+          this.buttons = {
+            singlebuttons: [
+              {
+                name: 'Apply Additional Shares',
+                icon: 'fa fa-arrow-right'
+              },
+              {
+                name: 'Redeem Shares',
+                icon: 'fa fa-arrow-left'
+              }
+            ],
+            options: [
+              {
+                name: 'Close'
+              }
+            ]
+          };
+        }
+
+        if (data.charges && data.charges.length !== 0) {
+          for (let i = 0; i < this.charges.length; i++) {
+            if (this.charges[i].name === 'Annual fee - INR') {
+              this.buttons.options.push({
+                name: 'applyAnnualFees'
+              });
+              this.annualChargeId = this.charges[i].id;
+            }
+          }
+        }
+      }
+    });
+  }
+
+  /**
+   * Gets Share Account Id from url params.
+   */
+  ngOnInit() {
+    this.route.params.subscribe(
+      params => {
+        this.shareAccountId = params['shareAccountId'];
+    });
+  }
+
+  /**
+   * For perform specific function according to the button clicked.
+   * @param eventName Name of the button clicked.
+   * @param accountId Share Account Id
+   */
+  clickEvent(eventName: any, accountId: any) {
+    eventName = eventName || '';
+    switch (eventName) {
+      case 'Modify Application':
+        this.router.navigate(['/editshareaccount/' + accountId]);
```

Comment:
Use parameters format. Do not concatenate.  Also, keep this in the same line as the case no need for extra lines. The file is already too long.